edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Serializer/DeflateSerializer.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Compression;
+    using System.Text;
+
+    /// <summary>
+    /// Simple wrapper for the System.IO.Compression.DeflateStream class to make it work on byte arrays.
+    /// </summary>
+    public static class DeflateSerializer
+    {
+        public static byte[] Compress(IEnumerable<byte> bytes)
+        {
+            using (MemoryStream compressedStream = new MemoryStream())
+            {
+                using (Stream compressionStream = new DeflateStream(compressedStream, CompressionLevel.Optimal, true))
+                {
+                    foreach (byte b in bytes)
+                    {
+                        compressionStream.WriteByte(b);
+                    }
+                }
+
+                return compressedStream.ToArray();
+            }
+        }
+
+        public static byte[] Decompress(IEnumerable<byte> bytes)
+        {
+            using (MemoryStream decompressedStream = new MemoryStream())
+            {
+                using (Stream decompressionStream = new DeflateStream(new IEnumerableReadStream(bytes), CompressionMode.Decompress, false))
+                {
+                    decompressionStream.CopyTo(decompressedStream);
+                }
+
+                return decompressedStream.ToArray();
+            }
+        }
+    }
+
+    class IEnumerableReadStream : Stream
```

Comment:
why this class named with "I" prefix; it is not an interface?  What does this class do?

Reply:
It was a stream that wrapped an IEnumerable. It isn't needed anymore, since I switched to the existing compression class.